airbyte-config/config-models/src/main/java/io/airbyte/config/Configs.java

```diff
@@ -541,13 +559,73 @@ public interface Configs {
    */
   boolean shouldRunConnectionManagerWorkflows();
 
+  /**
+   * Define if the worker is operating as a Control Plane worker, a Data Plane worker, or as a
+   * combined worker. - Control plane workers process tasks related to control-flow, like scheduling
+   * and routing. - Data plane workers process tasks related to data syncing, and are typically
+   * deployed outside of Airbyte's internal network. - Combined workers process all tasks, and are the
+   * default for OSS deployments. - Internal-use only.
+   */
+  WorkerPlane getWorkerPlane();
+
+  // Worker - Control Plane configs
+
+  /**
+   * Return true if the WorkerPlane is CONTROL_PLANE or COMBINED. Derived from the result of
+   * {@link Configs#getWorkerPlane()}. Internal-use only.
+   */
+  boolean isControlPlaneWorker();
+
+  /**
+   * TEMPORARY: Define a set of connection IDs that should run in Airbyte's AWS Data Plane. - This
+   * should only be set on Control-plane workers, since those workers decide which Data Plane task
+   * queue to use based on connectionId. - Will be removed in favor of the Routing Service in the
+   * future. Internal-use only.
+   */
+  Set<String> connectionIdsForDataPlane();
+
+  // Worker - Data Plane configs
+
+  /**
+   * Return true if the WorkerPlane is DATA_PLANE or COMBINED. Derived from the result of
+   * {@link Configs#getWorkerPlane()}. Internal-use only.
+   */
+  boolean isDataPlaneWorker();
```

Comment:
2 things:
1. I think it is weird that we don't just have a `getDeploymentType` (or equivalent name, I know deployment already means something else) that just returns the `WorkerPlane`. I'm not sure why we are doing this extra parsing of the meaning of this enum in this layer of the code.
2. `ControlPlaneWorker` is weird to me from a naming point of view. Is it really a "worker"? I don't think we have ever called the control plane a worker previously. I actually don't know if this is referring to a temporal queue running in the control plane or if it is just referring to the whole control plane.

Reply:
Continued in child PR here: https://github.com/airbytehq/airbyte/pull/15903/files#r953196019